test/test_nn.py

```diff
@@ -1556,6 +1556,90 @@ def test_module_to_argparse(self):
         with self.assertRaises(TypeError):
             net.to(cpu, torch.tensor(3, dtype=torch.long), non_blocking=True)
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    def test_global_flag_overwrite_module_params_on_conversion(self):
+        torch.__future__.set_overwrite_module_params_on_conversion(False)
+
+        # yf225 TODO: add comment for this test! Saying that the current default behavior is just wrong
+        m = nn.Linear(2,3)
+        mw = m.weight[:]
+        m.to('cuda')
+        mw[0][0] = 5
+        with self.assertRaisesRegex(RuntimeError, "Expected object of backend CPU but got backend CUDA"):
+            self.assertTrue(mw[0][0] == mw._base[0][0])
+
+        torch.__future__.set_overwrite_module_params_on_conversion(True)
+
+        # yf225 TODO: add comment for this test! Saying that the future behavior (aka. not doing cpu-cuda inplace) is correct
+        m = nn.Linear(2,3)
+        mw = m.weight[:]
+        m.to('cuda')
+        mw[0][0] = 5   # yf225 TODO: fix comment: We have a view on the old thing, and that needs to be consistent
+        self.assertTrue(mw[0][0] == mw._base[0][0])
+
+        # Test that when `torch.__future__.get_overwrite_module_params_on_conversion()`
+        # is true, `cpu_module.to("cuda")` doesn't preserve previous references
+        # to `cpu_module`'s parameters or gradients.
+        m = nn.Linear(20, 10)
+        m.weight.grad = torch.randn(10, 20)
+        weight_ref = m.weight
+        weight_grad_ref = m.weight.grad
+        m.to('cuda')
+        self.assertNotEqual(weight_ref.device, m.weight.device)
+        self.assertNotEqual(weight_grad_ref.device, m.weight.grad.device)
+
+        def add_one_inplace(t):
+            with torch.no_grad():
```

Comment:
why does this need to be in a no_grad block?  Shouldn't apply put it in a no-grad block anyway?

Reply:
Fixed.